cairo/ethereum/cancun/trie.cairo

```diff
@@ -1217,6 +1768,36 @@ func _get_preimage_for_key{poseidon_ptr: PoseidonBuiltin*}(
     return res;
 }
 
+// @notice Given a key (inside `dict_ptr`), returns the bytes32 preimage of the key registered in the tracker.
+// The preimage is validated to be correctly provided by the prover by hashing it and comparing it to the key.
+// @param key - The key to get the preimage for. Either a hashed or non-hashed key - but it must be a felt.
+// @param dict_ptr_stop - The pointer to the end of the dict segment, the one registered in the tracker.
+func _get_bytes32_preimage_for_key{poseidon_ptr: PoseidonBuiltin*}(
+    key: felt, dict_ptr_stop: DictAccess*
+) -> Bytes32 {
+    alloc_locals;
+
+    // Get preimage data
+    let (local preimage_data: felt*) = alloc();
+    local preimage_len;
+    %{ get_preimage_for_key %}
+
+    // Verify preimage
+    if (preimage_len != 2) {
+        with_attr error_message("_get_bytes32_preimage_for_key: preimage_len != 2") {
+            assert 0 = 1;
```

Comment:
`raise`

Reply:
instead changed to `assert preimage_len = 2`

Reply:
used raise